File: mkadia-backend/app/Models/Promotion.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Promotion extends Model
{
    use HasFactory;

    protected $fillable = [
        'code',
        'title',
        'description',
        'type',
        'value',
        'min_purchase_amount',
        'usage_limit',
        'usage_count',
        'is_active',
        'start_date',
        'end_date',
        'image_url'
    ];

    protected $casts = [
        'start_date' => 'datetime',
        'end_date' => 'datetime',
        'is_active' => 'boolean',
        'value' => 'decimal:2',
        'min_purchase_amount' => 'decimal:2',
    ];

    public function isValid()
    {
        $now = now();
        return $this->is_active &&
               $now->greaterThanOrEqualTo($this->start_date) &&
               $now->lessThanOrEqualTo($this->end_date) &&
               ($this->usage_limit === 0 || $this->usage_count < $this->usage_limit);
    }
}

File: mkadia_frontend/lib/models/promotion.dart
class Promotion {
  final String code;
  final String title;
  final String description;
  final String type;
  final double value;
  final double minPurchaseAmount;
  final int usageLimit;
  final int usageCount;
  final bool isActive;
  final DateTime startDate;
  final DateTime endDate;
  final String imageUrl;

  Promotion({
    required this.code,
    required this.title,
    required this.description,
    required this.type,
    required this.value,
    required this.minPurchaseAmount,
    required this.usageLimit,
    required this.usageCount,
    required this.isActive,
    required this.startDate,
    required this.endDate,
    required this.imageUrl,
  });

  factory Promotion.fromJson(Map<String, dynamic> json) {
    return Promotion(
      code: json['code'],
      title: json['title'],
      description: json['description'],
      type: json['type'],
      value: double.parse(json['value'].toString()), // Conversion explicite
      minPurchaseAmount: double.parse(json['min_purchase_amount'].toString()), // Conversion explicite
      usageLimit: json['usage_limit'],
      usageCount: json['usage_count'],
      isActive: json['is_active'],
      startDate: DateTime.parse(json['start_date']),
      endDate: DateTime.parse(json['end_date']),
      imageUrl: json['image_url'],
    );
  }
}